index.js

```diff
@@ -889,10 +889,8 @@ let createContainer = ({resolvers = [], dependencies = {}, factories, definition
 
         if (!definition.reuse) {
             destroyObject(instance, options);
-            definition.instance = null;
-        } else {
-            delete definition.instance;
         }
+        definition.instance = null;
```

Comment:
I could be wrong, but perhaps this is the correct behaviour?
Suppose that we have module `A`, and modules `B: '!A'` and `C: '!A'` in the same di session.
If in the next session I don't need `C` anymore but still need `B`, then I should not destroy `A` as it still in reuse.
Correct me please if I am wrong.